lib/package.gi

```diff
@@ -159,6 +159,159 @@ InstallGlobalFunction( SetPackageInfo, function( record )
     end );
 
 
+#############################################################################
+##
+#F  FindPackageInfosInSubdirectories( pkgdir, name )
+##
+##  Finds all PackageInfos in subdirectories of directory name in
+##  directory pkgdir, return a list of their paths.
+##
+BindGlobal( "FindPackageInfosInSubdirectories", function( pkgdir, name )
+    local pkgpath, file, files, subdir;
+    
+    if name in [ ".", ".." ] then
+      return [];
+    fi;
+
+    pkgpath:= Filename( [ pkgdir ], name );
+
+    # This can be 'fail' if 'name' is a void link.
+    if pkgpath = fail then
+      return [];
+    fi;
+
+    if not IsDirectoryPath( pkgpath ) then
+      return [];
+    fi;
+
+    file:= Filename( [ pkgdir ],
+                      Concatenation( name, "/PackageInfo.g" ) );
+    if file = fail then
+      files := [];
+      # Perhaps some subdirectories contain `PackageInfo.g' files.
+      for subdir in Set( DirectoryContents( pkgpath ) ) do
+        if not subdir in [ ".", ".." ] then
```

Comment:
Optional: I'd prefer `if subdir in [ ".", ".." ] then continue; fi;` (don't mind if it is on one line or three, the important bit is: I don't have to parse quite as deeply nested code in my brain)